File: JobApplicationTracker.Api/DTOs/JAStatusEntryDto.cs
using JobApplicationTracker.Enums;

namespace JobApplicationTracker.DTOs;

public class JAStatusEntryDto
{
    public Guid Id { get; set; }
    public Guid JobApplicationId { get; set; }
    public int OrderIndex { get; set; }
    public JAStatusType JaStatusType { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string? Note { get; set; }
}

File: JobApplicationTracker.Frontend/types/JAObjects/JAStatuses/JAStatusEntryDto.ts
import type { JAEventDto } from '../JAEvents/JAEventDto';

export interface JAStatusEntryDto {
  id: string;
  jobApplicationId: string;
  orderIndex: number;
  jaStatusType: number;
  createdAt: string;
  note?: string | null;
  jaEvent?: JAEventDto;
}